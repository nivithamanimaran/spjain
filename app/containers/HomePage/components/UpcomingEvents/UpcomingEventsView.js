import React from 'react';
import { Box, Typography, makeStyles, Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import EventRow from './EventRow';

const useStyles = makeStyles(theme => ({
  title: {
    fontSize: 14,
    fontWeight: 600,
    fontFamily: 'Work Sans'
  },
  container: {
    maxHeight: '50vh',
    minHeight: '45vh',
    overflowY: 'scroll',
    overflowX: 'hidden',
    '&::-webkit-scrollbar': {
      width: '4px',
    },
    '&::-webkit-scrollbar-track': {
      opacity: '0.2',
    },
    '&::-webkit-scrollbar-thumb': {
      backgroundColor: `#787878`,
      opacity: '0.1',
      outline: `1px solid #787878`,
    },
  },
  semiTitle: {
    color: theme.palette.KenColors.neutral400,
    fontSize: 12,
    marginLeft: 12,
    '&:hover': {
      cursor: 'pointer',
    },
  },
}));

const UpcomingEventsView = props => {
  const { events, handleSelected } = props;
  const styles = useStyles();
  const { t } = useTranslation();
  return (
    <div className={styles.container} data-testid={'upcoming-events'} style={{padding: '0px'}}>
      <Grid container direction="row" justify="space-between">
        <Grid item>
          <Typography className={styles.title}>
            {t('headings:Upcoming_Events')}
          </Typography>
        </Grid>
        <Grid item>
          <Grid container direction="row">
            <Typography
              data-testid="filter-all"
              className={styles.semiTitle}
              onClick={e => {
                handleSelected(e, 'all');
              }}
            >
              {t('labels:All')}
            </Typography>
            <Typography
              data-testid="filter-registered"
              className={styles.semiTitle}
              onClick={e => {
                handleSelected(e, 'registered');
              }}
            >
              {t('labels:Registered')}
            </Typography>
            <Typography
              data-testid="filter-unregistered"
              className={styles.semiTitle}
              onClick={e => {
                handleSelected(e, 'unregistered');
              }}
            >
              {t('labels:UnRegistered')}
            </Typography>
          </Grid>
        </Grid>
      </Grid>
      {events && (
        <Box mt={1} data-testid="events-container">
          {events?.map(event => (
            <EventRow event={event} />
          ))}
        </Box>
      )}
    </div>
  );
};

export default UpcomingEventsView;
