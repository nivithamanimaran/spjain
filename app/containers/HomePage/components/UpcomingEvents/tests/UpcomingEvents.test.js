import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import '@testing-library/jest-dom/extend-expect';
import ReactDOM from 'react-dom';
import ThemeProvider from '../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';
import UpcomingEventsView from '../UpcomingEventsView';

const events = [
  {
    Description:
      'PTM for Middle School will be taking place Online. All parents are requested to be available at allocated time slots. Please ensure that your ward is present for the meeting in the video call.',
    EndDateTime: '2021-06-26T21:30:00.000Z',
    Id: '00U1y0000034bWIEAY',
    Location: 'Online via Board Room, First Floor',
    StartDateTime: '2021-06-26T17:00:00.000Z',
    Subject: 'PTM- Middle School',
    count: 7,
    type: 'Un-Registered',
  },
  {
    Description: 'Upcoming MUN details have been sent to all Parents.',
    EndDateTime: '2021-07-07T18:00:00.000Z',
    Id: '00U1y0000034cVMEAY',
    Location: 'Online, Zoom Call',
    StartDateTime: '2021-07-07T17:00:00.000Z',
    Subject: 'MUN - Model United Nations',
    count: 3,
    type: 'Un-Registered',
  },
  {
    Description: 'Workshop on VR by Google',
    EndDateTime: '2021-06-30T11:07:13.000Z',
    Id: '00U1y0000034cwSEAQ',
    Location: 'Auditorium 1',
    StartDateTime: '2021-06-30T11:07:13.000Z',
    Subject: 'Workshop in VR',
    count: 1,
    type: 'Registered',
  },
  {
    Description: 'Description helloo ada asdas',
    EndDateTime: '2021-07-07T07:39:43.000Z',
    Id: '00U1y0000034djvEAA',
    Location: 'Delhi',
    StartDateTime: '2021-07-02T07:39:43.000Z',
    Subject: 'Test Subject',
    count: 0,
    type: 'Registered',
  },
];
const handleSelected = jest.fn((e, type) => {});

afterEach(() => {
  cleanup();
});
describe('Upcoming events test', () => {
  const div = document.createElement('div');
  it('should render and match the snapshot', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <UpcomingEventsView events={events} handleSelected={handleSelected} />
        </Router>
      </ThemeProvider>,
      div
    );

    const { findByTestId } = render(
      <ThemeProvider>
        <Router>
          <UpcomingEventsView events={events} handleSelected={handleSelected} />
        </Router>
      </ThemeProvider>
    );

    expect(findByTestId('events-container')).toMatchSnapshot();
  });
  it('Component should render without any error if events value is null', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <UpcomingEventsView events={null} handleSelected={handleSelected} />
        </Router>
      </ThemeProvider>,
      div
    );
  });
  it('filter handler should be called with proper args', async () => {
    const { getByLabelText, getByTestId } = render(
      <ThemeProvider>
        <Router>
          <UpcomingEventsView events={events} handleSelected={handleSelected} />
        </Router>
      </ThemeProvider>
    );
    const all = getByTestId('filter-all');
    expect(all).toBeTruthy();
    fireEvent.click(all);
    let firstArg = handleSelected.mock.calls[0][0];
    let secondArg = handleSelected.mock.calls[0][1];
    expect(Object.keys(firstArg).includes('target')).toBeTruthy();
    expect(secondArg).toBe('all');

    const registered = getByTestId('filter-registered');
    expect(registered).toBeTruthy();
    fireEvent.click(registered);
    firstArg = handleSelected.mock.calls[1][0];
    secondArg = handleSelected.mock.calls[1][1];
    expect(Object.keys(firstArg).includes('target')).toBeTruthy();
    expect(secondArg).toBe('registered');

    const unregistered = getByTestId('filter-unregistered');
    expect(unregistered).toBeTruthy();
    fireEvent.click(unregistered);
    firstArg = handleSelected.mock.calls[2][0];
    secondArg = handleSelected.mock.calls[2][1];
    expect(Object.keys(firstArg).includes('target')).toBeTruthy();
    expect(secondArg).toBe('unregistered');
  });

  it('events should be displayed with title, type and description', async () => {
    const { getAllByTestId } = render(
      <ThemeProvider>
        <Router>
          <UpcomingEventsView events={events} handleSelected={handleSelected} />
        </Router>
      </ThemeProvider>
    );
    const titles = getAllByTestId('event-title');
    expect(titles.length === 4).toBeTruthy();
    const tags = getAllByTestId('event-type-tag');
    expect(tags.length === 4).toBeTruthy();
    const descriptions = getAllByTestId('event-description');
    expect(descriptions.length === 4).toBeTruthy();

    expect(titles[0].textContent).toBe(events[0].Subject);
    expect(tags[0].textContent).toBe(events[0].type);
  });

  it('Description should have 70 characters max', async () => {
    const { getAllByTestId } = render(
      <ThemeProvider>
        <Router>
          <UpcomingEventsView events={events} handleSelected={handleSelected} />
        </Router>
      </ThemeProvider>
    );
    for (let i = 0; i < 2; i++) {
      let description = getAllByTestId('event-description')[i].textContent;
      let lastThreeChars = description.slice(
        description.length - 3,
        description.length
      );
      if (events[0].Description.length > 70) {
        expect(lastThreeChars).toBe('...');
      } else {
        expect(description).toBe(events[i].Description);
        expect(lastThreeChars).not.toBe('...');
      }
    }
  });
});
