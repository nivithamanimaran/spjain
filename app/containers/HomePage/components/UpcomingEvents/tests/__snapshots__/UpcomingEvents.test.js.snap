// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upcoming events test should render and match the snapshot 1`] = `Promise {}`;
